chia/consensus/block_body_validation.py

```diff
@@ -127,10 +145,21 @@ def include_block(
         self.update_fork_peak(block, header_hash)
         if block.foliage_transaction_block is not None:
             timestamp = block.foliage_transaction_block.timestamp
+            spent_coins: dict[bytes32, Coin] = {}
             for spend in removals:
-                self.removals_since_fork[bytes32(spend.name())] = ForkRem(bytes32(spend.puzzle_hash), block.height)
+                spend_id = bytes32(spend.name())
+                spent_coins[spend_id] = spend
+                self.removals_since_fork[spend_id] = ForkRem(bytes32(spend.puzzle_hash), block.height)
             for coin, hint in additions:
-                self.additions_since_fork[coin.name()] = ForkAdd(coin, block.height, timestamp, hint, False)
+                same_as_parent = (
+                    coin.amount % 2 == 1
+                    and coin.parent_coin_info in spent_coins
+                    and coin.puzzle_hash == spent_coins[coin.parent_coin_info].puzzle_hash
+                    and coin.amount == spent_coins[coin.parent_coin_info].amount
+                )
```

Comment:
this logic also needs to be the same as all other places where this flag is set. Here you check whether amount is odd, you don't above.